cli/src/package.rs

```diff
@@ -1,36 +1,115 @@
-use std::{env::current_dir, fs, path::PathBuf};
-
+use crate::error::CliError;
+use async_trait::async_trait;
 use directories::ProjectDirs;
+use serde_json;
+use std::{
+    env::current_dir,
+    fs::{self, create_dir_all, File},
+    io::copy,
+    path::PathBuf,
+};
 
+#[async_trait]
 pub trait Resolve {
```

Comment:
Move to Core and maybe remove `async`, you could just block before sending it to the resolve function. Also add a `resolve_all` function that resolves multiple paths in parallel